.github/actions/constellation_recover/action.yml

```diff
@@ -0,0 +1,102 @@
+name: Constellation recover
+description: "Recover a Constellation cluster with an unavailable control plane."
+
+inputs:
+  kubeconfig:
+    description: "The kubeconfig for the cluster."
+    required: true
+  masterSecret:
+    description: "The master-secret for the cluster."
+    required: true
+  cloudProvider:
+    description: "Which cloud provider to use."
+    required: true
+  gcpProject:
+    description: "The GCP project Constellation is deployed in."
+    required: false
+  resourceGroup:
+    description: "The Azure resource group Constellation is deployed in."
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    # Get Constellation UID
+    - name: Get Constellation cluster UID
+      shell: bash
+      run: |
+        CONSTELLATION_UID=$(yq -r ".uid" constellation-id.json)
+        echo "CONSTELLATION_UID=$CONSTELLATION_UID" >> $GITHUB_ENV
+      env:
+        KUBECONFIG: ${{ inputs.kubeconfig }}
+    # Restart worker node machine and wait for it to rejoin the cluster
+    - name: Restart worker node on GCP
```

Comment:
Have you explored some options for cloud provider independent restarts?

My first idea would be to [use `kubectl debug`](https://constellation-docs.edgeless.systems/constellation/workflows/troubleshooting#connect-to-nodes) and execute a reboot from the node itself. Do you think this could work and remove CSP specific implementation?

Reply:
@daniel-weisse what do you think? Is there any particular reason for not doing so on the worker node restart?

Reply:
I couldn't get the reboot of the host machine to work from inside a pod. Since the time left until 2.4.0 release is running out, I think we should leave this as is for now, while I will create a separate ticket to find a solution for this after the release. I think we should prefer having the test with csp-specific code over not having a test, but please let me know what you think here